packages/loaders/src/Dots.js

```diff
@@ -28,13 +28,13 @@ export default function Dots({
   delayMS = 750,
   ...other
 }) {
-  const { delayComplete } = useSchedule({ duration, delayMS });
   const noAnimatedSVGSupport = useCSSSVGAnimation();
+  const { delayComplete } = useSchedule({ duration, delayMS, loop: noAnimatedSVGSupport });
   const dotOne = useRef(null);
   const dotTwo = useRef(null);
   const dotThree = useRef(null);
 
-  useEffect(() => {
+  useLayoutEffect(() => {
```

Comment:
`useLayoutEffect` will [cause a warning to be shown when it's server-side rendered](https://reactjs.org/docs/hooks-reference.html#uselayouteffect). This hook is synchronous so the animation may look better, but the rest of the app could see some performance hits while this animation is running, especially with how often `delayComplete` is updated.

Did you attempt to use `React.memo` for the `Dots*Circle` components and `StyledSvg` element? Wrapping these elements in a `memo` seemed to have a larger impact in the profiler results.

Reply:
My primary concern is with the noise caused by repeated re-renderings in react devtools. The actual component's rendering isn't slow at all (<0.1ms).

Reply:
> My primary concern is with the noise caused by repeated re-renderings in react devtools

Wrapping the `Dots*` and `StyledSvg` component in a `React.memo(...)` would clean up the profiler flamegraph a lot for you. Since we're using the `ref`'s to apply the transform property I think this would have some good performance impacts as well.